internal/sqldb/sqlquery/cache.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sqlquery
+
+import (
+	"database/sql"
+	"log"
+
+	"github.com/datacommonsorg/mixer/internal/util"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+const (
+	SVGCacheKey = "c/cache/svgs"
+)
+
+// GetCacheData gets cache data for the specified key from the sql DB.
+// If not found, returns false.
+// If found, unmarshals the value into the specified proto and returns true.
+func GetCacheData(sqlClient *sql.DB, key string, out protoreflect.ProtoMessage) (bool, error) {
```

Comment:
no need for the bool return value. In this pattern, "nil" error signals a success

Reply:
Here we need to distinguish between an error and a value not found. If the out message was just a standard return type, a `nil` there would suffice. But since it is passed as a method parameter instead, the found / not found `bool` is the way to determine existence.

One option I considered was to return the `sql.ErrNoRows` error if a value was not found. But that felt like breaking encapsulation so went with a `bool` instead.

Reply:
Is it possible to set out to nil if no data is found?

Reply:
That won't work. For out params, we can only update the value, not change the reference.